### DISCUSSION TITLE
2024-05-24 - Employee KPIs - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by bradystroud** [SSWConsulting/SSW.Rules.Content#8592] [Chewing the Fat]
https://www.ssw.com.au/rules/employee-kpis/

During your annual review, its important to focus on the most important metrics first, and hopefully it will still enable you to get feedback and work out the big-ticket items for the following 12 months.

People are much more than statistics, KPIs are a good conversation starter.

What's your favorite KPI and why?

**Comment by PothieuG** [comment]
My favorite one is "Helping others".
You can't build anything great alone and without help. Every help or advise are gold.

**Comment by tkapa** [comment]
Helping Others - If you can (somehow) quantify this (I don't find Checked Bys to be indicative of the whole story) it would be great.

> People are much more than statistics, KPIs are a good conversation starter.

I hope that this holds true :)

**Reply by piers-sinclair** [reply]
Hey @tkapa 

The idea is that all the KPIs are subjectively measured and the source is just a guide point.

So when the KPIs are collated, the reviewer is going to give their own 2c on helping others beyond just checked bys. 

For example, they might know that on a certain project they don't use checked bys, but they know that person helps other people out a lot, so they would still give a high score.

**Comment by duanxinhuan** [comment]
My mom runs a accounting company and I have seen hundreds of companies from smaller to medium scale.
I know what's happening

**Comment by calumjs** [comment]
I think that naturally turning it into a point system will encourage people to maximise their score rather than doing necessarily what is best for the company.
e.g. it seems like I could improve my KPI by doing a lot less client work (utilisation is only worth up to 1), so I can free up a bunch more time to max out all those non-billable 0.25 weighting scores - exams, videos, updating rules, intranet, sugarlearning, sprint review emails, ctfs (worth a potential 1.75)
If everyone does this it will mean less utilisation, which I don't think is good.

I think I would probably keep it as a private internal number (and weighting system) just to give you a rough idea of how people are doing, but not make it public and not make it the only thing that matters.

**Reply by jakebayliss** [reply]
You always have great comments Calum

**Reply by AntPolkanov** [reply]
I think making the weighting system public is crucial as it clearly demonstrates the priorities and avoids people spending an excessive amount of time on nice-to-haves.

**Reply by SamBWagner** [reply]
Completely agree. If writing random Rules while on client work and doing worse work for clients means that my "number" is better. I'd be quite literally shooting myself in the foot to not do that. 

Thankfully integrity stops me from doing so. But, very bad to incentivise that kind of behaviour.

**Reply by wicksipedia** [reply]
Transparency is 👑
I'm waiting for the release notes when certain KPIs are nerfed/buffed

**Comment by tiagov8** [comment]
I'd say Spec Reviews because they bring us clients, but in fact each of the KPIs is important in its own way... the weights in the image seem right. 

Maybe I would give the 2 "Prospect" a 0.5 weight though. And move "Client Love" next to them

**Comment by matt-goldman-ssw** [comment]
It's awesome that we're providing transparency into this process here. Although KPIs are by definition measurable, so the qualitative KPIs are by definition not KPIs. See: https://www.kpi.org/kpi-basics/

> **Key Performance Indicators (KPIs)** are the critical (key) _quantifiable_ [emphasis mine] indicators of progress toward an intended result.

**Reply by leahy268** [reply]
Yes perhaps OKR's are more useful here :) https://www.microsoft.com/en-us/microsoft-viva/what-is-okr-objective-key-results

Some things are definitely hard to put down as a KPI though.  How do you measure 'Sentiment'.

**Reply by Hona** [reply]
Also - progress towards an intended result.
What result are we measuring towards?

**Reply by jaydenalchin** [reply]
+1 - I like that so many of these are qualitative but seems like those qualitative "KPIs" like client sentiment (arguably the most important one) are recorded as quantitative ratings/scores/metrics on a scale which is ultimately subjective or even arbitrary. 

IMO they should be pass/fail goals with notes - "Matt is a pleasure to have in class" - which is inherently harder for employees to gamify.

**Reply by SamBWagner** [reply]
Couldn't agree more. If we replaced KPIs with data in a study, that study would be laughed out of the building due to subjectivity and poor sample sizing alone. The emphasis should be on the word "Key", meaning that they hold weight and merit. All of these "metrics" are better suited to a coin toss. The assessment would probably track on to reality more closely.

**Comment by awaltos** [comment]
Client utilisation and satisfaction are super important. I also think helping with Business Development is important. I most like the one that tracks training/upskilling effort because they lead to better client work.

**Comment by danielmackay** [comment]
Helping others - I enjoy working with others to help them grow and learn.

There should be a KPI for being a 'thought leader'.  This would include blog posts, speaking at user groups or conferences, and running Superpowers/Workshops.

I like the fact that these are qualitative, and not just a pure measurement of statistics.

I also wonder, this seems very geared towards developers.  What about other staff like TV Team, Marketing, Managers?

**Reply by Hona** [reply]
Good idea & thoughts Dan.

**Reply by SamBWagner** [reply]
I agree with that. At the very least, total number of tries to get "out there", i.e. do videos, do conferences, etc is at the very least, measurable. I would still hold that there are too many confounding factors to actually glean meaningful value from it though.

The question I ALWAYS ask is one of exclusivity. If, given these KPIs over actually talking to someone or having any information on them. Could you make an informed decision. If the answer is no, they just aren't valuable and we should focus more on what is.

**Reply by kikibianc** [reply]
Agree - other teams would have some different KPIs, specially SysAdmins. There probably should be a template for each team.

**Comment by ncn-ssw** [comment]
My favorite KPI is the "KPI 2 - Sentiment - Clients" as it is most likely to lead to further work.

Having had to perform performance reviews in the past, I've found KPIs critical in providing structure and justification for the debrief.

**Comment by GabbyWard** [comment]
My favourite KPI is KPI 2 - Sentiment - Clients. 

While getting the work done and meeting expectations are important, ensuring their is a good standing relationship with the client before/during and after the work is key to laying the ground work for continued support and work, a client more willing to turn over competitors for ideas/work/suggestions and an easier time when working with the client.

**Comment by joshbermanssw** [comment]
My favourite KPIs tend to be related to 'sentiment - x'. I strongly believe that created relationships, and upholding a positive reputation, is a driver of future work. No-one wants to work with someone they dislike.

**Comment by vladislav-kir** [comment]
As consultants, "KPI 2 - Sentiment - Clients", in my opinion, is the most important KPI, since it shows how we grow relationships with the clients and reputation as a company.

**Comment by ricksu978** [comment]
I like  "Sentiment - Clients"

If Clients love our work, that's going to keep them coming back and could even help bring in new business through word-of-mouth.

**Comment by zacharykeeping** [comment]
"Sentiment - Clients" seems like probably the most important for consultants since it's how we keep and get new work.

**Comment by jimmy1992abc** [comment]
I like the system, but the weight can be examined to be more practical. 
E.g., Clients related may wroth more, and Internal or personal rate (State Managers/ Review Managers) could be less.

**Comment by Tanya-Leahy** [comment]
My favourite KPI is "Helping others".  I am not sure though how this could be "measured". We can learn so much from others, and in turn share our knowledge to create and achieve great things.

**Comment by TobyChurches** [comment]
I think 'Client - Sentiment' is the most important for a consultant, building a relationship where the Client trusts you and is happy with your work is **most** important. Depending on the client, this would likely be a lead to further sales

**Reply by Calinator444** [reply]
I'm just going to upvote this because I'd say that's my favourite KPI as well. When you do Client work you're not just generating revenue for the company, you're representing SSW as a brand.

**Comment by jakebayliss** [comment]
I think Sentiment - Clients is the most valuable and directly relates to more work and how easy we are to work with :)

**Comment by Hona** [comment]
It's a good idea to have KPIs be public, maybe even as a self-evaluation tool so that employees can get a gauge for what the boss is looking for (e.g. client work, update rules, upsell).

But a lot of these are very hard to measure accurately & in a way that is comparable to other employees.
It's also easy to game (e.g. 'please take me off client work, I want to be on the bench full time. I will check every email, keep my inbox at 0, create CTFs, update internal content').

**Comment by steven0x51** [comment]
Good to have KPIs, in my opinion Sentiment - Clients is the most valuable.
The current weighting of 0.25 for 'nice to have' KPIs is a bit high and risks overshadowing more critical KPIs.

**Comment by Ben0189** [comment]
"Sentiment - Clients" is the most valuable as a consultant 
strongly agree with "Helping others" as a valuable KPI internally

**Comment by arig2024** [comment]
"KPI 2 - Sentiment - Clients". 
 Client is King. 
No Clients = No business = No job.

**Comment by stephanfakossw** [comment]
Like that this is more efficient now 😊
Last year's goals - this one to me seems the most relevant. 
It can spawn many valuable conversations about personal and professional achievements.

**Comment by AntPolkanov** [comment]
"Exams"
It's a good quantitative metric, and external factors have the least influence on it.

**Comment by SamBWagner** [comment]
❌ I do not like any of these KPIs ❌

All of these KPIs are either out of the hands of the person being measured: Billability, Spec reviews, Initial meetings, etc

Or

Far too subjective and do not pass the most basic of scientific study standards. They have a tiny sample size, they're too biased, they're too subject to external factors, they're not able to be measured effectively, etc.

While I like the fact that we're trying to make the annual review process more streamlined, this is not the way to do it. This will lead to a more hazy experience, poorer outcomes and an overall less accurate assessment of value to the business.

**Reply by AntPolkanov** [reply]
None of the KPIs are perfect, but together they form a good framework.

Can you please give a couple of examples of metrics which are 
- measurable
- solely in the hands of the person being measured
- provide "_accurate assessment of value to the business_"
 
I think it's not that easy to come up even with three of those.

**Comment by uly1** [comment]
If this speeds things up... it's a huge win!
We'll have to figure out the variants for the tv, marketing, sales, management, finance, admin, and sysadmins parts of the company.

**Comment by jeoffreyfischer** [comment]
My favourite KPI: Helping others, because it creates a healthier work environment in which we all feel supported and part of the same team.
I also enjoy this CTF because it provides transparency on what is important to the company - it helps knowing what to focus on as an SSW consultant

**Comment by AttackOnMorty** [comment]
I prefer OKR with short cycles so we can get continuous feedback and adjust our goals as needed. 

These "KPIs" are something nice to have a look at.

**Comment by kikibianc** [comment]
My favourite KPI is "Efficiency" - one that we don't have!
Qualitative KPIs are cool but it's also important to measure raw numbers - PBIs, tickets, other numbers would give greater insight on how we're all doing.
I also think there should be better systems in place to gather feedback about each other anonymously, and more frequently. 
We could also expand on the "Peers" part, where everyone is ranked against each other to truly, fully show an individual's contribution inside the company. 
My feedback comes from seeing other companies' systems, especially Valve, where they have a fair review process that benefits everyone. I'll leave Valve's [employee handbook](https://steamcdn-a.akamaihd.net/apps/valve/Valve_NewEmployeeHandbook.pdf), and an image of how a stack ranking would work:
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/5884297/9b5707db-b189-4413-8bb3-4f2a175c050e)

**Reply by tomek-i** [reply]
> PBIs, tickets, other numbers would give greater insight on how we're all doing.

I disagree with this. Person A finishes 10 PBI's and Person B finishes 2 PBI's on paper looks good, what if I tell you Person A's PBI's were fixing spelling mistakes that took 10 minutes each while Person B actually wrote unit tests, removed technical debt and had way more impact to the project.

there is too much nuance in those numbers that are ALWAYS ignored by the reviewers and like you said they only look at the RAW numbers which are completely misleading and not indicative at all.

**Comment by amankumarrr** [comment]
KPI 2 - Sentiment - Clients

Being a good consultant means delivering what your client wants and meeting their expectations. This can be challenging, but we should always strive to give our best and ensure client satisfaction.

**Comment by JackDevAU** [comment]
+1 to Uly's comment. I think if this can make the annual review process shorter and smoother, i'm all for it! 
Having said that - I think as long as these KPIs aren't heavily relied on to a point where it becomes toxic this is great!

**Comment by MarcusIrmscherSSW** [comment]
My favourite KPI is sentiment, but I also think that many of these "KPI's" are very subjective and difficult to measure.

**Comment by chrisschultzssw** [comment]
In the form I said **Exams** is my favourite... pls everyone do exams 🥺
But really, **Helping others** is my favourite.

**Comment by cvionnet** [comment]
Having transparency is so great and important because it helps people understand what is important for the company and how they are assessed.

A challenging aspect is to fill in these KPIs regularly throughout the year and avoid doing so only before the annual review. This will help provide a fairer and more accurate view of the employee's performance.

There is also value for employees in filling out KPIs to review the company, as it can help detect areas for improvement.

**Comment by christoment** [comment]
I like the KPI related to collaborating with people since this also helps getting a better view on how an employee work from a different perspective.

However I think the biggest challenge is how to transform subjective measure into a 0-10 rating in a fair way. e.g. my "8" might indicate good enough while someone else's "8" might means way over the average - stellar performance.

**Comment by sethdaily** [comment]
Sentiment - clients
Unhappy clients = no business

**Comment by PennyWalker** [comment]
I do worry that this is going to be gamified... Developers are natural gamers and they work out really creative ways to make a system work for them and to hack it (looks at Rewards). I'm concerned a lot of these KPIs can be gamified and manipulated and that will be detrimental to the output we're trying to achieve. 

We do want to make Annual reviews shorter, without losing the value, but I'm not sure if this is the best way to measure people - perhaps it's a good kicking off point to start deeper conversations though.

**Comment by alexleessw** [comment]
Head says Sentiment - Clients
Heart says Helping others

**Comment by jerryluossw** [comment]
My favorite KPI is the **KPI 2 - Sentiment - Clients**, It's the most important for a consulting company.
At the same time, I believe we should not overlook the User Group.

**Comment by Aibono1225** [comment]
Sentiment - Clients
It's my favorite and I believe it's the most important one for a consulting company

**Comment by MattParkerDev** [comment]
I like the transparency and think this a good starting point. I love the conversation this has sparked and everyone getting involved in this discussion :)

I have seen many people say that `KPI 2 - Sentiment - Clients` is their favourite KPI, and I probably have to agree? If a client likes you, you must be doing something right - a combination of soft and hard skills that make a valuable consultant.

**Comment by jimmidier** [comment]
Sentiment - clients
The fundamental of a consulting company

**Comment by tomek-i** [comment]
Throughout my career, I’ve seen that management often says KPIs (Key Performance Indicators) are just discussion points and not strict performance measures. In reality, they often end up being used to judge someone's value or performance based on raw numbers. This can be misleading since these numbers rarely tell the whole story and can be quite subjective.

**Client Satisfaction**
Client satisfaction seems straightforward but can be distorted by unrealistic expectations. For example, clients with tight budgets may still expect top-tier results, leading to dissatisfaction that doesn't really reflect the team's or individual effort. 

**Helping Others / Teamwork**
Teamwork and helping others are key to a good work environment. Metrics like "checked by" instances might be helpful but don’t fully capture the various ways people contribute. Some roles involve more email communication, while others help in person. Between devs there are is a lot that goes undocumented like rubber duck debugging to name one.

**Individual Assessment**
Using a one-size-fits-all approach to KPIs isn't always effective. Everyone has unique strengths and roles that might not be captured by the same metrics. Comparing people directly can lead to unfair assessments. It makes more sense to evaluate each person based on their own growth and progress over the past year. This way, we recognize personal development and get more nuanced feedback from peers.

### Additional Concerns and Improvements

- **Subjectivity of Metrics**: Many KPIs are subjective and can be affected by factors outside an individual's control.
  - **Example**: A developer working on complex backend tasks might not produce as many visible features as a frontend developer but is still contributing significantly.

- **Quality Over Quantity**: Focusing on quantity-based metrics might lead to rushing work or prioritizing less impactful tasks.
  - **Example**: High-quality work and meaningful contributions, even if they take more time, should be encouraged.

- **Holistic Evaluation**: Combining qualitative assessments with quantitative KPIs can give a more balanced view of performance.
  - **Example**: Regular one-on-one meetings to discuss personal goals, challenges, and achievements can complement KPI data.

### Potential Solutions

- **Tailored KPIs**: Develop KPIs that are tailored to specific roles and responsibilities, acknowledging the diverse ways people contribute.
- **Growth and Development Focus**: Shift the focus from static metrics to individual growth and development, assessing how employees have progressed over time.
- **Peer Feedback**: Implement structured peer feedback mechanisms to gather insights on team dynamics and individual contributions that are not easily quantified.
- **Contextual Evaluation**: Consider the context in which metrics are achieved, recognizing external factors and providing a more nuanced interpretation of the data.

**Comment by KristenHu** [comment]
Client love - we should focus on clients definitely, that's what we're working for.
BTW, I really think different job levels should have different KPIs, solution architects should have different KPIs from regular engineers🤔

**Comment by babakamyljanovssw** [comment]
Sentiment - Clients is a important for consultant company and getting more work from clients

**Comment by jernejk** [comment]
My favorite one is helping others and mentoring. 😊

**Comment by miichaelsmedley** [comment]
Keeps everyone accountable - good to see SSW is trying to improve this process.

**Comment by zl24622** [comment]
I like all of the first 3 fundamental ones (Billable, Clients, and Helping others). 

If has to pick one, would be the My favorite one is KPI 1 - Clients - Billable

**Comment by KostaMadorsky** [comment]
I like that there is a balance in total score between between billable/clients and other KPIs.
So, if someone is mostly on client work, they may not be able to contribute much to other KPIs.
At the same time, if someone does not have a client for some period, they can score extra points in other KPIs.

**Comment by Marxoz** [comment]
If I have to pick one, Sentiment - Clients. Although that would be hard to measure.

**Comment by YalalCakirSSW** [comment]
Sentiment - Clients is the most important one among the list.

**Comment by bahjatalaadelssw** [comment]
I'm happy to give advice on a concise framework. I've done performance management in the past 😉

**Comment by tino-liu** [comment]
Sentiment - clients
For consulting, this is so important as it directly provides client retention and improves company reputation

**Comment by bettybondoc** [comment]
Top Pick: KPI 2 - Sentiment - Clients
Runner-up: Helping Others

Skills are important, but so is attitude. I’d always pick a good team player over a brilliant lone wolf. ✨

**Comment by ozair-ashfaque** [comment]
My favorite one is : Helping others
